pkg/controller/perconaservermongodb/pbm.go

```diff
@@ -188,6 +189,50 @@ func isResyncNeeded(currentCfg *config.Config, newCfg *config.Config) bool {
 	return false
 }
 
+func (r *ReconcilePerconaServerMongoDB) reconcilePiTRStorageLegacy(
+	ctx context.Context,
+	pbm backup.PBM,
+	cr *psmdbv1.PerconaServerMongoDB,
+) error {
+	log := logf.FromContext(ctx)
+	// if PiTR is enabled user can configure only one storage
+	var storage psmdbv1.BackupStorageSpec
+	for name, stg := range cr.Spec.Backup.Storages {
+		storage = stg
+		log.Info("Configuring PBM with storage", "storage", name)
+		break
+	}
+
+	var secretName string
+	switch storage.Type {
+	case psmdbv1.BackupStorageS3:
+		secretName = storage.S3.CredentialsSecret
+	case psmdbv1.BackupStorageAzure:
+		secretName = storage.Azure.CredentialsSecret
+	}
+
+	if secretName != "" {
```

Comment:
Maybe we can push this validation inside the `secretExists ` function, both for less nesting + for having consistency in case that function is reused.

Reply:
but in this case `secretExists` function needs to return `true` (exists) if secret name is empty

Reply:
In general `secretExists ` should not be called at all if the storage type is S3 or Azure. Providing to the `secretExists ` an empty name secret theoretically is a very good reason for this function to have a validation and throw an error. I understand that this may require some more changes, so we can skip it for now and revisit this when we touch it again.